src/arch/InputHandler/InputHandler_SDL.cpp

```diff
@@ -0,0 +1,974 @@
+        case SDL_GAMEPAD_BUTTON_MISC5:
+            return ssprintf("Misc 5 %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_MISC6:
+            return ssprintf("Misc 6 %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_TOUCHPAD:
+            return ssprintf("Touchpad %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_DPAD_UP:
+            return ssprintf("D-Pad Up %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_DPAD_DOWN:
+            return ssprintf("D-Pad Down %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_DPAD_LEFT:
+            return ssprintf("D-Pad Left %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_DPAD_RIGHT:
+            return ssprintf("D-Pad Right %s", deviceString);
+        default: break;
+        }
+
+        // Handle axis buttons (if any)
+        switch (axisBtn)
+        {
+        case SDL_GAMEPAD_AXIS_LEFT_TRIGGER:
+            if (isOlderSony || isNewerSony)
+            {
+                return ssprintf("L2 %s", deviceString);
+            }
+            if (isNintendo)
+            {
+                return ssprintf("ZL %s", deviceString);
+            }
+            return ssprintf("Left Trigger %s", deviceString);
+        case SDL_GAMEPAD_AXIS_RIGHT_TRIGGER:
+            if (isOlderSony || isNewerSony)
+            {
+                return ssprintf("R2 %s", deviceString);
+            }
+            if (isNintendo)
+            {
+                return ssprintf("ZR %s", deviceString);
+            }
+            return ssprintf("Right Trigger %s", deviceString);
+        case SDL_GAMEPAD_AXIS_LEFTX:
+            if (axisPositive)
+            {
+                return ssprintf("Left Stick Right %s", deviceString);
+            }
+            return ssprintf("Left Stick Left %s", deviceString);
+        case SDL_GAMEPAD_AXIS_LEFTY:
+            if (axisPositive)
+            {
+                return ssprintf("Left Stick Down %s", deviceString);
+            }
+            return ssprintf("Left Stick Up %s", deviceString);
+        case SDL_GAMEPAD_AXIS_RIGHTX:
+            if (axisPositive)
+            {
+                return ssprintf("Right Stick Right %s", deviceString);
+            }
+            return ssprintf("Right Stick Down %s", deviceString);
+        case SDL_GAMEPAD_AXIS_RIGHTY:
+            if (axisPositive)
+            {
+                return ssprintf("Right Stick Down %s", deviceString);
+            }
+            return ssprintf("Right Stick Up %s", deviceString);
+        default: break;
+        }
+
+        // Otherwise return the string
+        return ssprintf("%s", SDL_GetGamepadStringForButton(btn));
+    }
+    return InputHandler::GetDeviceSpecificInputString(di);
+}
+
+/**
+ * If the devices have changed notify the system.
+ * Note: The game will destroy and reinitialize this driver if this ever returns true.
+ * @return bool
+ */
+bool InputHandler_SDL::DevicesChanged()
+{
+    return m_devicesChanged;
```

Comment:
I built the device connection/disconnection code before realizing this caused itgmania to destroy and re-create this class. Is that really necessary?